sktime/transformations/series/temporian.py

```diff
@@ -0,0 +1,123 @@
+
+"""Implementation of TemporianTransformer. Contributed by the Temporian team."""
+
+__author__ = ["ianspektor"]  # TODO: add other Temporian devs who contributed to this
+
+from sktime.transformations.base import BaseTransformer
+
+
+# TODO: add usage example to docstring
+# see https://www.sktime.net/en/latest/developer_guide/dependencies.html#dependencies
+# for how to handle soft deps (temporian) there (need to not run doc tests on it)
+class TemporianTransformer(BaseTransformer):
+    """Applies a Temporian function to the input time series.
+
+    This transformer applies a [Temporian](https://temporian.readthedocs.io/en/stable/)
+    function to the input time series.
+
+    The conversion from sktime's internal representation to Temporian's
+    [EventSet](https://temporian.readthedocs.io/en/stable/reference/temporian/EventSet/)
+    and back is handled automatically by the transformer.
+
+    Parameters
+    ----------
+    function : Callable[[temporian.EventSet], temporian.EventSet]
+        Temporian function to apply to the input time series. The function must receive
+        and return a single Temporian EventSet, and can apply an arbitrary number of
+        Temporian operators to its input.
+    """
+
+    _tags = {
+        "univariate-only": False,
+        "authors": [
+            "ianspektor"
+        ],  # TODO: add other Temporian devs who contributed to this
+        "maintainers": [
+            "ianspektor"
+        ],  # TODO: add other Temporian devs who will maintain this
+        "scitype:transform-input": "Series",
+        "scitype:transform-output": "Series",
+        "scitype:instancewise": True,
+        "scitype:transform-labels": "None",
+        # TODO: extend to pd-multiindex and pd_multiindex_hier
+        "X_inner_mtype": "pd.DataFrame",
+        "fit_is_empty": True,
+        "capability:inverse_transform": False,
+        "capability:unequal_length": True,
+        "handles-missing-data": False,
+        "python_dependencies": ["temporian"],
+        "python_version": "<3.12",
+    }
+
+    def __init__(self, function):
+        self.function = function
+
+        super().__init__()
+
+        # TODO: ensure function receives a single EventSet/param with inspect module?
+        #       or is failing in runtime OK?
+        # TODO: @tp.compile the function
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            if self.get_tag("univariate-only")==True:
+                guaranteed to have a single column
+            if self.get_tag("univariate-only")==False: no restrictions apply
+        y : None, present only for interface compatibility
+
+        Returns
+        -------
+        transformed version of X
+        """
+        import temporian as tp
```

Comment:
In case of import failure, should there be a nice error message with instructions (e.g., pip install) on how to install Temporian if missing?

Reply:
I'm guessing this is handled by sktime already - @fkiraly?